### PR TITLE
feat: auth and api calls

### PR DESCRIPTION
**Pull request by bhtibrewal**
- basic-authentication added
- did some API calls

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
+import { SignIn, SignUp } from "./pages";
 
 
 function App() {
   return (
     <div className="body">
+      <Routes>
+<Route path='/sign-in' element={<SignIn/>}/>
+<Route path='/sign-up' element={<SignUp/>}/>
+        </Routes>
```
Comment (Ankur9669) [inline src/App.jsx]: fix: Indentation could be better

File: src/App.jsx [hunk 2]
```diff
@@ -1,12 +1,18 @@
 import "./App.css";
+import {Routes, Route} from 'react-router-dom';
 import {Toast} from './components'
+import { SignIn, SignUp } from "./pages";
 
 
 function App() {
```
Comment (Ankur9669) [inline src/App.jsx:6]: you can use es6

File: src/context/NotesContext.jsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+const NotesContext = createContext();
+
+export const NotesProvider = ({ children }) => {
+  const initialNotesState = {
+    notes: [],
+    archives: [],
+  };
+  const notesReducer = (state, { type, payload }) => {
+    switch (type) {
+      case "":
```
Comment (Ankur9669) [inline src/context/NotesContext.jsx:12]: fix: Why empty case

File: src/custom_hooks/useDocumentTitle.jsx [hunk 4]
```diff
@@ -2,6 +2,6 @@ import { useEffect } from "react";
 
 export const useDocumentTitle = (title = '') => {
   useEffect(() => {
```
Comment (Ankur9669) [inline src/custom_hooks/useDocumentTitle.jsx:4]: Suggestion: This is possible without useEffect also

File: src/pages/auth/sign_in/SignIn.jsx [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+      type: "primary",
+    });
+    return () => {};
+  }, []);
+
+  return (
+    <main className="main center">
+      <form
+        className="flex-col signup-sec"
+        onSubmit={(e) => {
```
Comment (Ankur9669) [inline src/pages/auth/sign_in/SignIn.jsx]: Suggestion: Can move this in separate function

File: src/pages/auth/sign_in/SignIn.jsx [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+          label={"Email"}
+        />
+        <PasswordInput
+          value={inputValues.password}
+          onChange={(e) =>
+            setInputValues({ ...inputValues, password: e.target.value })
+          }
+          label={"Password"}
+        />
+        <label className="flex-align-center">
```
Comment (Ankur9669) [inline src/pages/auth/sign_in/SignIn.jsx:72]: fix: use htmlFor here

File: src/pages/auth/sign_up/SignUp.jsx [hunk 7]
```diff
@@ -0,0 +1,129 @@
+import { useState } from "react";
+import {  useNavigate } from "react-router-dom";
+import { ButtonPrimary, InputField, PasswordInput } from "../../../components";
+import { useToast, useAuth } from "../../../context";
+import { signUp } from "../../../services";
+
```
Comment (Ankur9669) [inline src/pages/auth/sign_up/SignUp.jsx:6]: Can import all files into index.js if you want

File: src/pages/auth/sign_up/SignUp.jsx [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+  const emailPattern = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/;
+  const validEmail = emailPattern.test(email);
+
+  const canSubmit = () => {
+    return firstName !== "" && lastName !== "" && validEmail && validPassword;
+  };
+  /* if agree to terms and can submit then set disabled false */
+  const isDisabled = () => !(agreeToTerms && canSubmit());
+  const disabled = isDisabled();
+
```
Comment (Ankur9669) [inline src/pages/auth/sign_up/SignUp.jsx:47]: Appreciation: Nice use of regex in validation

File: src/pages/auth/sign_up/SignUp.jsx [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+  return (
+    <main className="main center">
+      <form
+        className="flex-col signup-sec"
+        onSubmit={(e) => {
```
Comment (Ankur9669) [inline src/pages/auth/sign_up/SignUp.jsx]: Same suggestion try to make a seprate function for this

**Review by Ankur9669 — APPROVED** [review]
The code looks nice to me, do check the suggestions else feel free to merge

File: src/context/NotesContext.jsx (hunk 3 above)
Comment (Mohitk391) [reply src/context/NotesContext.jsx:12]: Suggestion: if there is no action to be taken for the states in reducer then you should remove switch case as it results in no change in any action for now, if in future there occurs some state changes then you can provide reducer definition there..

Edit: Okay, I understand that why currently there is nothing in here as you have just declared these functionalities and not defined in this commit, which you will be providing in the future commit, as this commit is only for auth and api calls..

**Review by Mohitk391 — APPROVED** [review]
Appreciation: Clean, readable code and liked that you have added a bit more customized auth functionality. Apart from suggestions provided by Ankur, you can merge the commit.

File: src/context/NotesContext.jsx (hunk 3 above)
Comment (bhtibrewal) [reply src/context/NotesContext.jsx:12]: exactly mohit they are just defined now for future use

File: src/pages/auth/sign_up/SignUp.jsx (hunk 9 above)
Comment (bhtibrewal) [reply src/pages/auth/sign_up/SignUp.jsx]: Thanks Ankur for the suggestion, okay will do that

File: src/pages/auth/sign_up/SignUp.jsx (hunk 8 above)
Comment (bhtibrewal) [reply src/pages/auth/sign_up/SignUp.jsx:47]: Thank you

**Pull request merged**